File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/feedback/MentionCommentEntity.java
package dev.ragnarok.fenrir.db.model.entity.feedback;

import com.google.gson.annotations.SerializedName;

import dev.ragnarok.fenrir.db.model.entity.CommentEntity;
import dev.ragnarok.fenrir.db.model.entity.Entity;
import dev.ragnarok.fenrir.db.model.entity.EntityWrapper;
import dev.ragnarok.fenrir.model.feedback.FeedbackType;

/**
 * Base class for types [mention_comments, mention_comment_photo, mention_comment_video]
 */
public class MentionCommentEntity extends FeedbackEntity {

    @SerializedName("where")
    private CommentEntity where;
    @SerializedName("commented")
    private EntityWrapper commented = new EntityWrapper();

    @SuppressWarnings("unused")
    public MentionCommentEntity() {
    }

    public MentionCommentEntity(@FeedbackType int type) {
        setType(type);
    }

    public CommentEntity getWhere() {
        return where;
    }

    public MentionCommentEntity setWhere(CommentEntity where) {
        this.where = where;
        return this;
    }

    public Entity getCommented() {
        return commented.get();
    }

    public MentionCommentEntity setCommented(Entity commented) {
        this.commented = new EntityWrapper().wrap(commented);
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/feedback/MentionCommentEntity.kt
package dev.ragnarok.fenrir.db.model.entity.feedback

import androidx.annotation.Keep
import dev.ragnarok.fenrir.db.model.entity.CommentEntity
import dev.ragnarok.fenrir.db.model.entity.DboEntity
import dev.ragnarok.fenrir.model.feedback.FeedbackType
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Base class for types [mention_comments, mention_comment_photo, mention_comment_video]
 */
@Keep
@Serializable
@SerialName("mention_comment")
class MentionCommentEntity : FeedbackEntity {
    var where: CommentEntity? = null
        private set

    private var commented: DboEntity? = null

    @Suppress("UNUSED")
    constructor()
    constructor(@FeedbackType type: Int) {
        this.type = type
    }

    fun setWhere(where: CommentEntity?): MentionCommentEntity {
        this.where = where
        return this
    }

    fun getCommented(): DboEntity? {
        return commented
    }

    fun setCommented(commented: DboEntity?): MentionCommentEntity {
        this.commented = commented
        return this
    }
}